src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -249,15 +247,25 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var lineFormattingOptions = await document
                 .GetLineFormattingOptionsAsync(fallbackOptions: null, cancellationToken).ConfigureAwait(false);
             var modifiedClassTrivia = GetModifiedClassTrivia(
-                positionalParameterInfos.SelectAsArray(p => p.Declaration), typeDeclaration, lineFormattingOptions);
+                positionalParameterInfos, typeDeclaration, lineFormattingOptions);
 
-            var propertiesToAddAsParams = positionalParameterInfos.ZipAsArray(defaults, (p, @default) =>
-                SyntaxFactory.Parameter(
-                    GetModifiedAttributeListsForProperty(p.Declaration),
+            var propertiesToAddAsParams = positionalParameterInfos.Zip(defaults, (result, @default) =>
+            {
+                // if inherited we generate nodes and tokens for the type and identifier
+                var type = result.IsInherited
+                    ? result.Symbol.Type.GenerateTypeSyntax()
+                    : result.Declaration.Type;
+                var identifier = result.IsInherited
+                    ? SyntaxFactory.Identifier(result.Symbol.Name)
+                    : result.Declaration.Identifier;
```

Comment:
why do we not also do that if it's not inherited?

Reply:
I would like to use the identifier that the declaration uses if it's there, so we can keep trivia if there is any. But, if we're inherited, the declaration is in a different class as the parameter list, so we don't really want the trivia associated with that declaration, as the parameter we add just shadows the inherited property.